airflow/www/decorators.py

```diff
@@ -125,3 +127,29 @@ def wrapper(self, *args, **kwargs):
                                         __class__.__name__ + ".login"))
         return wrapper
     return decorator
+
+
+def log_webserver_stats(endpoint, log_views=False, log_duration=False, log_failures=False):
+    """
+    Decorator to log server side stats from webserver UI.
+
+    Currently supports logging views and load failures as counters
+    and load duration as timing.
+    """
+    def decorator(f):
+        @functools.wraps(f)
+        def wrapper(*args, **kwargs):
+            start_dttm = timezone.utcnow()
+            if (log_views):
+                Stats.incr('webserver.{}.views'.format(endpoint))
+            if (log_failures):
+                Stats.incr('webserver.{}.load_failures'.format(endpoint))
+            response = f(*args, **kwargs)
+            if (log_duration):
+                duration = (timezone.utcnow() - start_dttm).total_seconds() * 1000
+                Stats.timing('webserver.{}.load_time'.format(endpoint), duration)
+            if (log_failures):
+                Stats.decr('webserver.{}.load_failures'.format(endpoint))
```

Comment:
I have doubts whether increasing and decreasing the counter is the best solution. Can not the counter be increased only when an error has occurred?

Reply:
The idea around this is to catch silent errors in the implementation. We sometimes have errors that are more transient/unexpected, which causes endpoints to fail. In that case we might not hit the try catch to increment the failures count. This seemed like an assured way to catch any and all failures.

Reply:
This is going to do funny things to the stats over time.